### PR TITLE
Add ARP entries for RubyInstallerTeam.Ruby.3.0 version 3.0.4-1

### PR DESCRIPTION
**Pull request by russellbanks**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [ ] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`? 
- [ ] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.1 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.1.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/66864)

### CODE REVIEW

File: manifests/r/RubyInstallerTeam/Ruby/3/0/3.0.4-1/RubyInstallerTeam.Ruby.3.0.installer.yaml [hunk 1]
```diff
@@ -19,8 +19,14 @@ Installers:
...
   InstallerSha256: DAD3C96582DB658E4257BA92BC151D737CCA9F3BAF8ECA991B158BBAC39F4931
+  AppsAndFeaturesEntries:
+    DisplayName: Ruby 3.0.4-1-x64-msvcrt
+    ProductCode: RubyInstaller-3.0-x64-mingw32_is1
 - Architecture: x86
   InstallerUrl: https://github.com/oneclick/rubyinstaller2/releases/download/RubyInstaller-3.0.4-1/rubyinstaller-3.0.4-1-x86.exe
   InstallerSha256: F5E38D82828D43D3255712CF2690194607C5DBF361BDA48B3A84DAF781C599A4
+  AppsAndFeaturesEntries:
+    DisplayName: Ruby 3.0.4-1-x86-msvcrt
+    ProductCode: RubyInstaller-3.0-i386-mingw32_is1
```
Comment (OfficialEsco) [inline manifests/r/RubyInstallerTeam/Ruby/3/0/3.0.4-1/RubyInstallerTeam.Ruby.3.0.installer.yaml:30]:
```suggestion
  AppsAndFeaturesEntries:
  - DisplayName: Ruby 3.0.4-1-x86-msvcrt
    ProductCode: RubyInstaller-3.0-i386-mingw32_is1
```

File: manifests/r/RubyInstallerTeam/Ruby/3/0/3.0.4-1/RubyInstallerTeam.Ruby.3.0.installer.yaml [hunk 2]
```diff
@@ -19,8 +19,14 @@ Installers:
 - Architecture: x64
   InstallerUrl: https://github.com/oneclick/rubyinstaller2/releases/download/RubyInstaller-3.0.4-1/rubyinstaller-3.0.4-1-x64.exe
   InstallerSha256: DAD3C96582DB658E4257BA92BC151D737CCA9F3BAF8ECA991B158BBAC39F4931
+  AppsAndFeaturesEntries:
+    DisplayName: Ruby 3.0.4-1-x64-msvcrt
+    ProductCode: RubyInstaller-3.0-x64-mingw32_is1
```
Comment (OfficialEsco) [inline manifests/r/RubyInstallerTeam/Ruby/3/0/3.0.4-1/RubyInstallerTeam.Ruby.3.0.installer.yaml:24]:
```suggestion
  AppsAndFeaturesEntries:
  - DisplayName: Ruby 3.0.4-1-x64-msvcrt
    ProductCode: RubyInstaller-3.0-x64-mingw32_is1
```

**Review by OfficialEsco — CHANGES_REQUESTED** [review]
I can smell a automation crash

**Comment by russellbanks** [comment]
> I can smell a automation crash

I'm doing these manually; I'll just check the Add-ARP script does it correctly better aha

**Review by Trenly — APPROVED** [review]

**Pull request merged**